File: common/src/main/java/com/demo/common/dto/TaskCompletionStatus.java
package com.demo.common.dto;

/**
 * Completion state of a task based on its current phase.
 * Used as a filter parameter on list endpoints to find finished or dev-finished tasks.
 */
public enum TaskCompletionStatus {
    /** Task is in RELEASED or REJECTED phase — fully finished, all modifications blocked. */
    FINISHED,
    /** Task is in DONE phase — dev work complete, comments and booked work still allowed. */
    DEV_FINISHED
}

File: android/data/src/main/kotlin/com/demo/taskmanager/data/dto/enums/TaskCompletionStatus.kt
package com.demo.taskmanager.data.dto.enums

import kotlinx.serialization.Serializable

/** Filter param: FINISHED = RELEASED/REJECTED phase; DEV_FINISHED = DONE phase. */
@Serializable
enum class TaskCompletionStatus {
    FINISHED,
    DEV_FINISHED,
}
